### PR TITLE
Add solution for izho-17-LBS

### PR DESCRIPTION
**Pull request by rsj001**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

**Comment by rsj001** [comment]
That was weird, seemed like a lot of button in Preview were broken

File: solutions/izho-17-LongestBeautifulSequence.mdx [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+id: izho-17-LongestBeautifulSequence
+source: IZhO 2017
+title: 2017 - Longest beautiful sequence
+author: Shijie Ren
+---
+
+<Spoiler title="Hint 1">
+
+Both $\mathcal{O}(N^2)$ and $\mathcal{O}(N\max{a[i]})$ algorithms are not hard
+to come up with, but the latter one is more enlightening.
```
Comment (SansPapyrus683) [inline solutions/izho-17-LongestBeautifulSequence.mdx:11]: please elaborate
not everyone is as smart as you

Comment (SansPapyrus683) [reply solutions/izho-17-LongestBeautifulSequence.mdx:11]: also, what does "more enlightening" mean?

File: solutions/izho-17-LongestBeautifulSequence.mdx [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+You may find two ways of $\mathcal{O}(N\max{a[i]})$ DP.
+
+One is kind of preprocessing previous answers, traverse all the situations of
+the previous number, $\mathcal{O}(\max{a[i]})$ to query current answer in DP
+array and $\mathcal{O}(1)$ to update.
+
+The other is kind of predicting next answers, traverse all the possible outcomes
+of the next number, $\mathcal{O}(1)$ to query current answer in DP array and
+$\mathcal{O}(\max{a[i]})$ to update, but needs a bit more memory.
```
Comment (SansPapyrus683) [inline solutions/izho-17-LongestBeautifulSequence.mdx:25]: what's the DP state? how exactly do you transition?

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
start slow

File: solutions/izho-17-LongestBeautifulSequence.mdx (hunk 1 above)
Comment (rsj001) [reply solutions/izho-17-LongestBeautifulSequence.mdx:11]: ok, but they're repeated in hint2, and their equations are given in the explanation

Comment (rsj001) [reply solutions/izho-17-LongestBeautifulSequence.mdx:11]: > also, what does "more enlightening" mean?

I want to express n2 solution can't be optimized further, maybe "more enlightening" is meaningless

Comment (rsj001) [reply solutions/izho-17-LongestBeautifulSequence.mdx:11]: ; )

File: solutions/izho-17-LongestBeautifulSequence.mdx (hunk 2 above)
Comment (rsj001) [reply solutions/izho-17-LongestBeautifulSequence.mdx:25]: their details are in the explanation

Comment (SansPapyrus683) [reply solutions/izho-17-LongestBeautifulSequence.mdx:25]: move it here so it flows better then

File: solutions/izho-17-LongestBeautifulSequence.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/izho-17-LongestBeautifulSequence.mdx:11]: > ok, but they're repeated in hint2, and their equations are given in the explanation

then move them here, just saying that they exist does basically nothing

**Comment by rsj001** [comment]
anything else

**Comment by SansPapyrus683** [comment]
did you not see my discord dm

File: solutions/izho-17-LongestBeautifulSequence.mdx [hunk 3]
```diff
@@ -88,6 +87,23 @@ $$
 \end{align}
 $$
 
+And we want some number AND $\texttt{cur} = 8$.
```
Comment (SansPapyrus683) [inline solutions/izho-17-LongestBeautifulSequence.mdx:90]: 8? why 8?

Comment (rsj001) [reply solutions/izho-17-LongestBeautifulSequence.mdx:90]: oh it's not late to find , it should be 9

**Comment by rsj001** [comment]
nope

**Pull request closed without merging**